### ISSUE TITLE
[net::ERR_INVALID_CHUNKED_ENCODING] generic-session

### ISSUE THREAD

**Issue by d30jeff** [warmcat/libwebsockets#722]
Hi, some of the resources of my webapp failed to be loaded.

The front end that I'm using is built on Angular, there's a command that *minifies* the **css** and **js** files.

Anyway, it only seemed to break on **javascript** files, but not **css** files.

I noticed the header of those two files is different.

e.g.
**Response Header** for **css**
>accept-ranges:bytes
>cache-control:no-store
>connection:keep-alive
>content-length:112612
>content-type:text/css
>etag:0001B7E457E23AFC
>server:lwsws

**Response Header** for **javascript**
>accept-ranges:bytes
>cache-control:no-store
>connection:keep-alive
>content-type:text/javascript
>etag:000C045957E23AFC
>server:lwsws
>transfer-encoding:chunked

Anyways, I'm getting these errors:
![Errors](http://i.imgur.com/YEM0emW.jpg)

I suspected that the size of those files (**js** are bigger than **css**) affected this, but when I tried to add a dummy javascript file with very little content, it still breaks.

Could it be that it's because of libwebsockets not being able to load/ return the response of those script files, in one shot?

Is there a way to *increase* the size/limit?

Edit: Formatting

**Comment by d30jeff** [comment]
I also have tried adding `<script type="text/javascript" src="my-script.js"></script>` to the html.

**Comment by lws-team** [comment]
It looks like it's lws chunking it during the interpreter step (that replaces eg $lwsgs_user in .js).  We tell it to scan *.js to replace them.

Normally the file that has js in the messageboard demo is pretty small, maybe it has problems chunking beyond a certain size (there is no limitation in the code, but maybe some dumb bug).

Can you let me know the size of one of the .js guys having trouble?

**Comment by d30jeff** [comment]
The size of the `.js` file is **56.0K**

**Comment by lws-team** [comment]
just in case can you tell me the exact number of bytes

**Comment by d30jeff** [comment]
Idk how to `du`(Google isn't helping) so that it would return the size in bytes,
`ls -l` returns the following output:

>-rw-rw-r-- 1 deojeff deojeff  54910 Dec 16 18:51 scripts.dc41d154.js
>-rw-rw-r-- 1 deojeff deojeff 787545 Dec 16 18:51 vendor.29dd69df.js

So I guess, for **scripts.*.js** it's **54910** bytes.
For **vendor.*.js** it's **787545** bytes

**Comment by lws-team** [comment]
Yes ls -l is the way, thanks.

I'll study this later tonight after I finish my current deathmatch.

**Comment by lws-team** [comment]
It's the fault of the recent ranges support patch in master, it works itself but its integration into output.c trashed chunked encoding generally, it's nothing to do with big or small files.

It's fixed in master.

I fix things like this in master by just amending the responsible patch, if it makes it hard to follow it a good way to handle it is

 - git fetch git://github.com/warmcat/libwebsockets.git +master:m7a --tags

this zaps current master into a local branch "m7a" or whatever name you like, without disturbing anything in your current checkout.

If you then want to zap your current branch to be exactly the same as current master, you can do

 - git reset --hard m7a

or you can change to a new branch and check out m7a or whatever.  The point it this is fast and keeps everything in one repo instead of cloning into a new dir each time.

Or you can clone a new dir each time if you prefer.

**Comment by tejendrasingh** [comment]
sir,  chunked upload is possible on lws server? actually i used curl cmd for upload file on lws server with  
--header "Transfer-Encoding : chunked" --limit-rate 6000  . but nothing is happening .. curl did get reponse from server

**Comment by lws-team** [comment]
@tejendrasingh  open your own issue if you have a different question.

What exactly are you trying to do?

There are two meanings to "chunked" for upload.

1) http/1 chunked transfer encoding https://en.wikipedia.org/wiki/Chunked_transfer_encoding .  This bug is discussing that.

2) "sliced" concurrent uploads.  As far as I can tell, there is no "chunking" in the http sense for "chunked" / sliced file upload.  I can't find any RFC or other standard about it so it's hard to be sure.

If you literally mean uploads with chunked transfer encoding, no it's not supported atm.  Is there a reason to support it?

**Comment by lws-team** [comment]
@d30jeff this is solved by the patch?

**Comment by d30jeff** [comment]
@lws-team Yes, it's solved by the patch. Sorry for the late reply. Closing...
